docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,981 @@
+       102                                     pkt.mark == 42           allow
+```
+
+And these will be the gateway router configured routes (no policies needed):
+```
+IPv4 Routes
+Route Table <main>:
+           169.254.0.0/17               169.254.0.4 dst-ip rtoe-GR_test12_namespace.scoped_ovn-control-plane
+           203.203.0.0/16               100.88.0.13 dst-ip rtotr-GR_test12_namespace.scoped_ovn-control-plane <-- cluster ingress or egress reply traffic going towards the pod network via transit_router peer IP.
+                0.0.0.0/0                172.18.0.1 dst-ip rtoe-GR_test12_namespace.scoped_ovn-control-plane
+```
+
+For the routes related to EIP, they are moved from gateway router to transit_router and the redirect action is changed to point to the transit node peer ip.
+
+#### Router to router direct connection without a switch
+
+To connect the transit_router to the local gateway router instead of using a join switch like the layer3 topology those ports can be connected each other directly
+using the `peer` field.
+
+This is how it looks the `trtor` port at transit_router:
+```
+_uuid               : f697bce1-dac7-442d-9355-e298e1735c7b
+dhcp_relay          : []
+enabled             : []
+external_ids        : {"k8s.ovn.org/network"=test12_namespace-scoped, "k8s.ovn.org/topology"=layer2}
+gateway_chassis     : []
+ha_chassis_group    : []
+ipv6_prefix         : []
+ipv6_ra_configs     : {}
+mac                 : "0a:58:64:58:00:0d"
+name                : trtor-GR_test12_namespace.scoped_ovn-control-plane
+networks            : ["100.88.0.13/30", "fd97::8/127"]
+options             : {requested-tnl-key="4"}
+peer                : rtotr-GR_test12_namespace.scoped_ovn-control-plane <------------- peer field
+status              : {}
+```
+
+And this is the `rtotr` port at GR:
+```
+_uuid               : f9ab92f3-478c-41dd-b845-0d8ddf4a34e5
+dhcp_relay          : []
+enabled             : []
+external_ids        : {"k8s.ovn.org/network"=test12_namespace-scoped, "k8s.ovn.org/topology"=layer2}
+gateway_chassis     : []
+ha_chassis_group    : []
+ipv6_prefix         : []
+ipv6_ra_configs     : {}
+mac                 : "0a:58:64:41:00:04"
+name                : rtotr-GR_test12_namespace.scoped_ovn-control-plane
+networks            : ["100.65.0.4/16", "100.88.0.14/30", "fd97::9/127", "fd99::4/64"]
+options             : {gateway_mtu="1400"}
+peer                : trtor-GR_test12_namespace.scoped_ovn-control-plane <-------------- peer field
+status              : {}
+
+```
+
+#### Transit router specifics
+
+For transit router to work the LRPs referencing the same gateway router peer should have a unique tunnel key and if they are remote also a
+requested chassis pointing to the gateway router node.
+
+Also in the case we still need to support conditional SNAT, the transit router port connected to the switch needs to be configured as gateway router port.
```

Comment:
hmm can you expand on this part? Not sure I follow this... I thought the reason we needed to have that specific gateway router port was because of this whole topology mess so the second cSNAT on the GR side can now remove its match on gateway port right? for L2? cause now each router only has 1 cSNAT?

Reply:
ah OVN Gateway Routers! not OVN-Kubernetes Gateway routers.. yikes OK that one was a miss in my head that landed in the wrong contextual place :)